### PR TITLE
[release/3.1] Update a few more components

### PR DESCRIPTION
**Pull request by dougbu**
- address CG bugs w/ older versions of these `npm` components
- update `urijs` version in one project
- update `webpack` and related versions in a few projects
  - pin `chokidar` instead of `tar`

### CODE REVIEW

File: eng/SharedFramework.External.props [hunk 1]
```diff
@@ -114,6 +114,7 @@
     <_CompilationOnlyReference Include="System.ComponentModel.Annotations" />
     <_CompilationOnlyReference Include="System.Runtime.CompilerServices.Unsafe" />
     <_CompilationOnlyReference Include="System.Text.Json" />
+    <_CompilationOnlyReference Include="System.Text.Encodings.Web" />
```
Comment (dougbu) [inline eng/SharedFramework.External.props:117]: @wtgodbe due to package version conflicts between App.Ref and other projects, I suspect this is needed both here and below. Do you remember if that's correct❔

File: src/Components/Web.JS/package.json [hunk 2]
```diff
@@ -24,18 +24,17 @@
...
-    "webpack-cli": "^3.3.11"
+    "webpack": "^4.46.0",
+    "webpack-cli": "^3.3.12"
   },
   "resolutions": {
     "**/set-value": "^2.0.1",
-    "bl": ">=2.2.1",
+    "chokidar": ">=3.5.3",
     "lodash": ">=4.17.21",
-    "tar": ">=4.4.18",
```
Comment (dougbu) [inline src/Components/Web.JS/package.json:38]: If validation fails again, will probably revert or reduce the `webpack` version bump. We're using an older version in release/5.0 too.

**Comment by dougbu** [comment]
@dotnet/aspnet-blazor-eng could you please have a look at the changes in this PR❔

**Comment by wtgodbe** [comment]
I'm remembering now that we already dismissed the 5.0/6.0 versions of these alerts, since they apply only to `node_modules` files that aren't checked in to the repo & we trust that the version of `tar` we're using explicitly is the safe one. Will merge this once green & dismiss the `node_modules` alerts.

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**